### PR TITLE
:sparkles: testdata/cronjob_types.go: add NestedStructWithSeveralFieldsDoubleMarked

### PR DESCRIPTION
**Pull request by jan--f**
This adds a field to the cronjob test data, that ensures that fields
which are marked twice (once through a type marker, one through a field
marker) still generate the correct yaml structure.

Signed-off-by: Jan Fajerski <jfajersk@redhat.com>

### CODE REVIEW

**Comment by FillZpp** [comment]
I'm not sure, should we keep this test to verify that annotated on both field and type, to avoid bugs like these #688 and https://github.com/kubernetes-sigs/controller-tools/pull/693#issuecomment-1169433662

**Comment by jan--f** [comment]
> I'm not sure, should we keep this test to verify that annotated on both field and type, to avoid bugs like these #688 and [#693 (comment)](https://github.com/kubernetes-sigs/controller-tools/pull/693#issuecomment-1169433662)

Yes that makes sense, I added a commit to that effect. WDYT?

**Comment by jan--f** [comment]
ping @FillZpp Any further concerns about this?

**Comment by FillZpp** [comment]
@jan--f I don't understand what's the difference between `NestedStructWithSeveralFields` and `NestedStructWithSeveralFieldsDoubleMarked `?

**Comment by jan--f** [comment]
> @jan--f I don't understand what's the difference between `NestedStructWithSeveralFields` and `NestedStructWithSeveralFieldsDoubleMarked `?

@FillZpp Right, because I forgot to add the field marker. Added it now, thank you!

**Comment by jan--f** [comment]
@chrischdi anything still blocking this?

**Comment by chrischdi** [comment]
/lgtm

from my side. But requires a maintainer for approval (which I am not).

**Comment by jan--f** [comment]
Thanks, let me try @pwittrock and @mengqiy for an `approve` then.

File: pkg/crd/testdata/cronjob_types.go
Comment (camilamacedo86) [inline pkg/crd/testdata/cronjob_types.go:170]: It seems a workarround.  Could you please raise an issue or add in the description:

a) what you were doing when you face the error ( with an example code )
b) what do you expected to see ?
c) what was the behaviour faced instead? 

Also, if we move forward here I think we need a better name than `FieldsDoubleMarked`. DoubleMarket with what? (i.e. NestedStructWithSeveralFieldsMarkedWithAB)

File: pkg/crd/testdata/cronjob_types.go
Comment (camilamacedo86) [inline pkg/crd/testdata/cronjob_types.go:165]: Why it was removed?

**Review by camilamacedo86 — CHANGES_REQUESTED** [review]
It seems requires more info first.

File: pkg/crd/testdata/cronjob_types.go
Comment (jan--f) [reply pkg/crd/testdata/cronjob_types.go:170]: The context of this is https://github.com/kubernetes-sigs/controller-tools/issues/676, which is a duplicate of https://github.com/kubernetes-sigs/controller-tools/issues/435.

Comment (jan--f) [reply pkg/crd/testdata/cronjob_types.go:170]: @camilamacedo86 so this just adds an additional test case. `NestedStructWithSeveralFields` ensures that the resulting manifest carries the `atomic` marker from the type. `NestedStructWithSeveralFieldsDoubleMarked` ensures the same thing only that the marker is applied to the type, as well as the to the field of the struct.

**Review by alvaroaleman — APPROVED** [review]

**Comment by alvaroaleman** [comment]
Thanks!

**Pull request merged**
